### PR TITLE
Test Kotlin 1.6.21 and Kotlin 1.7.0 in CI.

### PR DESCRIPTION
**Pull request by vRallev**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -103,6 +103,12 @@ subprojects {
       substitute module("com.squareup.anvil:compiler-api") with project(':compiler-api')
       substitute module("com.squareup.anvil:compiler-utils") with project(':compiler-utils')
     }
+
+    // TODO: check if this is needed with newer Kotlin versions.
+    resolutionStrategy {
+      // Needed for Kotlin 1.7.
+      force deps.kotlin.metadata
```
Comment (vRallev) [inline build.gradle:110]: We use an older Dagger version and need to bump the transitive dependency.

File: settings.gradle [hunk 2]
```diff
@@ -6,7 +6,14 @@ include ':integration-tests:code-generator'
 include ':integration-tests:code-generator-tests'
 include ':integration-tests:dagger-factories-only'
 include ':integration-tests:library'
-include ':integration-tests:mpp:android-module'
+
+// TODO: There seems to be a bug with Kotlin 1.7 and MPP projects. Disable the Android
+//  module for now until the issue is resolved
+def kotlinVersion = hasProperty('square.kotlinVersion') ? getProperty('square.kotlinVersion').toString() : '1.6.10'
+if (VersionNumber.parse(kotlinVersion).baseVersion < VersionNumber.parse('1.7.0')) {
+  include ':integration-tests:mpp:android-module'
```
Comment (vRallev) [inline settings.gradle:14]: Not sure what's going on. For some reason code within `androidDebug` can't see code from `commonMain` with Kotlin 1.7.

File: settings.gradle [hunk 3]
```diff
@@ -6,7 +6,14 @@ include ':integration-tests:code-generator'
 include ':integration-tests:code-generator-tests'
 include ':integration-tests:dagger-factories-only'
 include ':integration-tests:library'
-include ':integration-tests:mpp:android-module'
+
+// TODO: There seems to be a bug with Kotlin 1.7 and MPP projects. Disable the Android
+//  module for now until the issue is resolved
```
Comment (0xMatthewGroves) [inline settings.gradle:11]: Ultra nit: extra space

**Review by 0xMatthewGroves — APPROVED** [review]

File: settings.gradle (hunk 3 above)
Comment (vRallev) [reply settings.gradle:11]: No, the extra space will make sure that IJ renders the multiline TODO as a single TODO.

**Review by JoelWilcox — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (ZacSweers) [reply build.gradle:110]: I believe dagger shades this since it's unstable? Could be wrong

**Pull request merged**